### DISCUSSION TITLE
Importer Action - CSV File Upload Issue on Firefox

### DISCUSSION THREAD

**Discussion by salvobee** [filamentphp/filament#10556] [Ideas]
I have encountered an issue related to the prebuilt action for importing records via CSV file upload.

When attempting to upload a CSV file using Firefox on Windows, I encountered a problem with the accepted MIME types by the form. The form indicated that it expected either "text/plain" or "text/csv." I attempted to create the CSV file by manually assembling some lines with Notepad++ and saving the file as plain text with a .csv extension. Additionally, I exported a file from Excel using the CSV format. However, regardless of these efforts, I was unable to upload any CSV file.

Upon further investigation, I used JavaScript to check the detected file type by my browser (Firefox) for these created files. Surprisingly, it was identified as "application/vnd.ms-excel." In contrast, Chrome detected them as "text/csv."

I delved into this issue and discovered that Firefox relies on the system registry to determine MIME types. I found a relevant discussion on the Mozilla support forum: [Mozilla Support Forum](https://support.mozilla.org/en-US/questions/1401889).

I understand that this behavior might be influenced by how my system is configured to handle these types of files. However, I believe it is important to highlight that users of an application utilizing this functionality may encounter difficulties uploading files. Therefore, it would be beneficial to explore a solution to handle this use case effectively.

**Comment by JakeVdub** [comment]
Thank you for posting this, I have been battling this issue for the past day or so. An application I developed with Filament was working fine on my machine, but not on my client's. Turns out, they were using Firefox and trying to upload a CSV file that they exported out of Microsoft Excel, which has the mime type of `application/vnd.ms-excel` as you noted. This is not an accepted file type of the Importer, and causes the upload to fail.
